### ISSUE TITLE
Terrain Flattening results are currently underperforming

### ISSUE THREAD

**Issue by maawoo** [SAR-ARD/S1_NRB#62]
https://forum.step.esa.int/t/inferior-terrain-flattening-results-when-using-external-dem/37221

**Comment by johntruckenbrodt** [comment]
Thanks for the great summary in this STEP issue. It was increasingly hard to follow the discussions around this topic. Let's hope they soon come up with a fix.

**Comment by johntruckenbrodt** [comment]
Things are moving forward:

https://forum.step.esa.int/t/terrain-flattening-results-in-garbage-when-using-copernicus-30m-auto-download/33777/34

> The reason that the TF result obtained using auto downloaded Copernicus 30m DEM is different from that obtained using external Copernicus 30m DEM file is that the DEM oversampling factor is computed differently. The oversampling factor is computed using the DEM resolution and the image pixel spacing. In the former case, the DEM resolution is given while in the latter case the DEM resolution is computed using Earth model. We have updated the oversampling factor calculation so that it is the same for both cases. The fix will be available in the next release.

**Comment by maawoo** [comment]
Finally! 🎉

**Comment by johntruckenbrodt** [comment]
https://github.com/SAR-ARD/S1_NRB/pull/78 should bring some improvement. To be revised once the new version is out.

**Comment by johntruckenbrodt** [comment]
https://github.com/senbox-org/s1tbx/releases/tag/9.0.3

**Comment by johntruckenbrodt** [comment]
New consolidated forum post about the current status:  
https://forum.step.esa.int/t/external-dem-handling-in-snap-consolidated-findings-on-geocoding-terrain-flattening-pixel-alignment-and-vertical-datum/46074
